### PR TITLE
Update DeveloperGuide.md

### PR DESCRIPTION
**Pull request by chia-yh**
* Remove [Proposed] from implementation of undo/redo
* Rename fields/methods to current implementation
* Credit original proposed implementation used in AB-3
* Update UC for redo

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,41 +154,41 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Undo/redo feature
 
-#### Proposed Implementation
+This feature was implemented as proposed in [AddressBook-Level3](https://se-education.org/addressbook-level3/DeveloperGuide.html#proposed-undoredo-feature) 
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The proposed undo/redo mechanism is facilitated by `VersionedSocket`. It extends `AddressBook` with an undo/redo history, stored internally as an `socketStateList` and `currentStatePointer`. Additionally, it implements the following operations:
```
Comment (chia-yh) [inline docs/DeveloperGuide.md:161]: `AddressBook` intentionally left unchanged until refactoring complete

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -384,7 +382,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
     * 3a1. SOCket shows an error message.
 
-      Use case resumes from step 2.
+      Use case resumes at step 2.
```
Comment (chia-yh) [inline docs/DeveloperGuide.md:385]: standardized to resumes **at** instead of **from**

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -594,7 +592,7 @@ Use case ends.
 
 * 3a. The contact does not have github information.
     
-   Use case resumes from step 2.
+   Use case resumes at step 2.
 
 **Use case: UC13 Undo recent delete/clear operation**
```
Comment (chia-yh) [inline docs/DeveloperGuide.md:597]: @huanghao1998 , should we consider removing steps 1 and 2, as they conflict with Extension 3a? I added a similar UC for redo with that in mind, however, if you feel like it is necessary I will change UC14 to follow the format for UC13

Comment (huanghao1998) [reply docs/DeveloperGuide.md:597]: Yes I agreed!! We should remove step 1 and 2 and follows the UC14

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -646,7 +661,7 @@ Use case ends.
 
     * 5a1. SOCket shows an error message.
   
-      Use case resume from 6 when the date entered are correct.
+      Use case resume at step 6 when the date entered is correct.
```
Comment (huanghao1998) [inline docs/DeveloperGuide.md:664]: Thanks for helping me out with my grammatical error

**Review by huanghao1998 — APPROVED** [review]
LGTM!

**Comment by chia-yh** [comment]
Honestly no clue why codecov is failing when no changes were made to functional code

**Review by huanghao1998 — APPROVED** [review]
Nice to merge some of the use cases!

**Pull request merged**